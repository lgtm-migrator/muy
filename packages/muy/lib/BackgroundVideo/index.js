import React from "react"
import Box from "@material-ui/core/Box"
import Video from "../Video"
import makeStyles from "@material-ui/core/styles/makeStyles"

const useStyles = makeStyles({
  root: {
    transform: "translate(-50%, -50%)",
    "@media screen and (max-aspect-ratio: 1920/1080)": {
      height: "100%",
    },
    "@media screen and (min-aspect-ratio: 1920/1080)": {
      width: "100%",
    },
  },
})

const BackgroundVideo = ({ props }) => {
  const classes = useStyles()
  return (
    <Box
      className={"MuyBackgroundVideoContainer"}
      height={"100vh"}
      left={0}
      overflow={"hidden"}
      position={"fixed"}
      top={0}
      width={"100vw"}
      zIndex={-1}
    >
      <Video
        autoPlay={true}
        // bottom={0}
        className={classes.root}
        display={"block"}
        left={"50%"}
        loop={true}
        // margin={"auto"}
        minHeight={"50%"}
        minWidth={"50%"}
        muted={true}
        overflow={"hidden"}
        position={"absolute"}
        // right={0}
        top={"50%"}
        zIndex={1}
        {...props}
      />
    </Box>
  )
}

export default BackgroundVideo
