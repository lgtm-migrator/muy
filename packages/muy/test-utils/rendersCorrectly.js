import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
