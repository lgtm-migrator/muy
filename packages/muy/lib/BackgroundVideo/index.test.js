import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import BackgroundVideo from "./index"

describe("<BackgroundVideo />", () => {
  it("should render correctly", () => {
    rendersCorrectly(<BackgroundVideo />)
  })
})
