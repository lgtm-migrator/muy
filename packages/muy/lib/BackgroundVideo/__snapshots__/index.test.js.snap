// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundVideo /> should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 MuyBackgroundVideoContainer"
>
  <video
    autoPlay={true}
    className="MuiBox-root MuiBox-root-3 makeStyles-root-1"
    loop={true}
    muted={true}
  />
</div>
`;
